test/smartwallet/nativeHolderSmartWallet.test.ts

```diff
@@ -1,401 +1,398 @@
-                senderPrivateKey,
-                relayRequest,
-                chainId
-            );
-
-            await nativeSw.execute(
-                suffixData,
-                relayRequest.request,
-                worker,
-                signature,
-                { from: worker }
-            );
-
-            // @ts-ignore
-            const logs = await recipientContract.getPastEvents(
-                'SampleRecipientEmitted'
-            );
-            expect(logs.length).to.be.equal(0, 'TestRecipient should not emit');
-        });
+  }
+
+  async function expectBalanceAfterSuccessExecution(
+    recipientAddress: string,
+    value: BigNumber,
+    execution: () => Promise<void>
+  ) {
+    const recipientBalancePriorExecution = await provider.getBalance(
+      recipientAddress
+    );
+    const swBalancePriorExecution = await provider.getBalance(
+      smartWallet.address
+    );
+
+    const valueBN = BigNumber.from(value);
+
+    const expectedRecipientBalance = BigNumber.from(
+      recipientBalancePriorExecution
+    ).add(valueBN);
+    const expectedSwBalance = BigNumber.from(swBalancePriorExecution).sub(
+      valueBN
+    );
+    await expectBalanceToBeRight(
+      execution,
+      recipientAddress,
+      expectedRecipientBalance,
+      expectedSwBalance
+    );
+  }
+
+  describe('execute', function () {
+    type PrepareRequestParams = {
+      data: string;
+      value?: string;
+    };
+
+    function prepareRequest({
+      data = '0x',
+      value = TWO_ETHERS.toString(),
+    }: PrepareRequestParams) {
+      const relayRequest = createRequest(
+        {
+          relayHub: relayHub.address,
+          from: owner.address,
+          to: recipientContract.address,
+          tokenAmount: '10',
+          tokenGas: '40000',
+          tokenContract: fakeToken.address,
+          data,
+          value,
+        },
+        {
+          callForwarder: smartWallet.address,
+        }
+      );
+
+      const typedRequestData = new TypedRequestData(
+        HARDHAT_CHAIN_ID,
+        smartWallet.address,
+        relayRequest
+      );
+      const signature = getLocalEip712Signature(typedRequestData, privateKey);
+      const suffixData = getSuffixData(typedRequestData);
+
+      return {
+        relayRequest,
+        suffixData,
+        signature,
+      };
+    }
+
+    it('Should transfer native currency without executing transactions', async function () {
+      const { relayRequest, suffixData, signature } = prepareRequest({
+        data: '0x',
+        value: TWO_ETHERS.toString(),
+      });
+
+      const execution = async () => {
+        await expect(
+          smartWallet
+            .connect(relayHub)
+            .execute(
+              suffixData,
+              relayRequest.request,
+              worker.address,
+              signature
+            )
+        ).not.to.be.rejected;
+
+        expect(fakeToken.transfer, 'Token.transfer() was not called').to.be
```

Comment:
I think this assert is redundant since having the correct balance in each side, is a guarantee that this function was called.